### DISCUSSION TITLE
How to setup ufw/ufwg for t-pot

### DISCUSSION THREAD

**Discussion by xgithx** [telekom-security/tpotce#1673] [Q&A]
Considering [T-Pot System Placement](https://github.com/telekom-security/tpotce?tab=readme-ov-file#system-placement)

Which are the commands to configure Ubuntu firewall from the terminal, in order for the T-pot to work properly. Or maybe, should I try any other firewalls for Ubuntu, more than ufw/ufwg, so I can use T-pot, as it should?

**Comment by xgithx** [comment]
I have tried Google, but no result. Maybe with ufw/ufwg to not be possible to set up those rules? What alternative to ufw/ufwg do you recommend, just to work those rules?

**Reply by dmille6** [reply]
i'm in the same spot fighting the same thing.. there is a rules.sh in one of the tools folders that is run. it has something to do with the tpot firewall.. but even modifying it doesn't seem to give me the results I want.. (I want to lock down 64295-64298 to a specific IP address)

if you find a solution let me know.

**Reply by xgithx** [reply]
Like [(https://youtu.be/sbZwMO6mjPE?t=1381&si=Zu675ZDTB6HS0aMA)](url) from the minute 23, I must add those two rules in ufw/gufw, because I don't use aws, like in the clip. How to do it? I have attached a print screen with gufw, if in this print screen should be possible to add those two rules, or, if someone knows how to
![20241014_065709](https://github.com/user-attachments/assets/cd45e550-681b-4745-9a7e-30e255015ff3)
cmd terminal sudo command to add these two rules.

**Comment by xgithx** [comment]
I want to set up through gufw or through ufw directly from the terminal, the next settings from the first print screen

The second print screen is the one from gufw, if in that gufw window, I must set up these rules.
![2](https://github.com/user-attachments/assets/11416fae-2ae4-4d8e-a41f-cbd360d6e934)
![20241014_065709](https://github.com/user-attachments/assets/1645d20b-2763-47c0-9213-faa96bf81ccf)

**Comment by dmille6** [comment]
i've done this from the terminal and it doesn't effect the tpot.. i think it has something to do with the docker interfaces that are created. but I haven't seen anyone in the discussions find a way to do this.

**Comment by xgithx** [comment]
It seems that for the first rules the command is:

`sudo ufw allow from 141.0.145.180/32 to any port 64297 proto tcp`

Unfortunately, for the second rule, it should be something like that, but it doesn't work:



```
sudo ufw allow 0:64000/tc
ERROR: Bad Port.​
```


Any ideas, please?
Thank you.

**Reply by dmille6** [reply]
try sudo ufw allow 1:64000/tc

but ive tried similar things.. everything sets up correctly but traffic still isnt accepted and blocked like it should
let me know if you get it working.
